save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/PermissionController.kt

```diff
@@ -61,16 +61,24 @@ class PermissionController(
     )
     fun getRole(@PathVariable organizationName: String,
                 @PathVariable projectName: String,
-                @RequestParam userName: String,
+                @RequestParam(required = false) userName: String?,
                 authentication: Authentication,
-    ): Mono<Role> = permissionService.findUserAndProject(userName, organizationName, projectName)
+    ): Mono<Role?> = permissionService.findUserAndProject(
+        // fixme: userName should be like that: ${user.source}:${user.name}
+        userName ?: authentication.name.split(":")[1],
```

Comment:
There is `id` and `identitySource` only in `AuthenticationDetails`. That is a temporary solution because later we will need not only `userName`, but also a `source` in order to tell gh users from gl users.

Reply:
We have a (apprarently, unused) method `Authentication.toUser` in `SecurityUtils.kt`. Can it be used here? So that logic is not duplicated.

Reply:
Oh yes, good. Thanks

Reply:
Do we need to ping database if we have `Authentication.toUser()`? Cause as far a I can see, it also provides user's role.

Reply:
It's the global role, which is read from the DB once the server receives the first request with this authentication. It's different from per-project roles, and is already present in the `Authentcation` object

Reply:
ok then, will get username from `Authentication.toUser()`